Code/Framework/AzToolsFramework/Tests/Prefab/Benchmark/Spawnable/SpawnAllEntitiesBenchmarks.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#if defined(HAVE_BENCHMARK)
+
+#include <Prefab/Benchmark/Spawnable/SpawnableBenchmarkFixture.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+#include <AzToolsFramework/Prefab/Spawnable/SpawnableUtils.h>
+
+namespace Benchmark
+{
+    using BM_SpawnAllEntities = BM_Spawnable;
+
+    BENCHMARK_DEFINE_F(BM_SpawnAllEntities, SingleEntitySpawnable_SpawnCallVariable)(::benchmark::State& state)
+    {
+        const unsigned int spawnAllEntitiesCallCount = static_cast<unsigned int>(state.range());
+        const unsigned int entityCountInSourcePrefab = 1;
+
+        CreateSpawnable(entityCountInSourcePrefab);
+
+        for (auto _ : state)
+        {
+            m_spawnTicket = new AzFramework::EntitySpawnTicket(m_spawnableAsset);
```

Comment:
Repeated: Should there a be `state.PauseTiming();` and `state.ResumeTiming();` around this call as well? This because it now spends time on the memory allocation for the ticket, which doesn't typically happen in real-world scenarios plus the benchmark is for spawning and not so much for creating tickets.

Reply:
That's a fair call out. We can have separate benchmarks around creating a ticket if needed later. Excluded this from time calculations now.